xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp

```diff
@@ -171,8 +171,10 @@ bool CEGLNativeTypeAmlogic::ProbeResolutions(std::vector<RESOLUTION_INFO> &resol
   RESOLUTION_INFO res;
   for (std::vector<std::string>::const_iterator i = probe_str.begin(); i != probe_str.end(); ++i)
   {
-    if(aml_mode_to_resolution(i->c_str(), &res))
-      resolutions.push_back(res);
+    if (((StringUtils::StartsWith(i->c_str(), "4k2k")) && (aml_support_h264_4k2k() > AML_NO_H264_4K2K)) || !(StringUtils::StartsWith(i->c_str(), "4k2k"))) {
```

Comment:
Questions:
- what is aml_support_h264_4k2k() requesting? hardware? A static list anywhere around?
- EGLNativeTypeAmlogic is used for Odroid C2 / S905 as well, where is this check for this?

Coding style:
- Pls put brackets in a new line, thx!